From 978459358758b3bfefa366d71d444ed9e093793a Mon Sep 17 00:00:00 2001
From: Meron Cohen <mecohen@habana.ai>
Date: Wed, 07 May 2025 21:39:27 +0300
Subject: [PATCH] [SW-227605] g2e: support hl225e variant

Define a new eFuse bit (bit #27 of address 0x3a) for HL225E
(BIS-E-COMPLAINT) and define a new FIT load type for creating
a new .itb file for writing this bit to the eFuse.

Change-Id: I8acc0cd33f8bff250cab3f7a597671921555e377
Signed-off-by: Meron Cohen <mecohen@habana.ai>
---

diff --git a/fit_load_type.h b/fit_load_type.h
index 5bc4fa3..eb28596 100644
--- a/fit_load_type.h
+++ b/fit_load_type.h
@@ -50,7 +50,8 @@
 	FIT_LOAD_TYPE_COPY_EFUSE_TO_FAKE_EFUSE = 37,
 	FIT_LOAD_TYPE_READ_LOG = 38,
 	FIT_LOAD_TYPE_WRITE_RACK_EEPROM = 39,
-	FIT_LOAD_TYPE_COUNT = 40,
+	FIT_LOAD_TYPE_BIS_E_COMPLIANT_ENABLE = 40,
+	FIT_LOAD_TYPE_COUNT = 41,
 };
 
 #endif /* FIT_LOAD_TYPE_H */
diff --git a/gaudi2/gaudi2_security.h b/gaudi2/gaudi2_security.h
index a0d907a..8da26e5 100644
--- a/gaudi2/gaudi2_security.h
+++ b/gaudi2/gaudi2_security.h
@@ -76,9 +76,12 @@
 #define EFUSE_ENABLE_BIS_COMPLIANT		BIT(25)
 /* EFUSE enable BIS rev D compliant */
 #define EFUSE_ENABLE_BIS_D_COMPLIANT		BIT(26)
+/* EFUSE enable BIS rev E compliant */
+#define EFUSE_ENABLE_BIS_E_COMPLIANT		BIT(27)
 
 #define EFUSE_BIS_COMPLIANT_MASK		(EFUSE_ENABLE_BIS_COMPLIANT | \
-						 EFUSE_ENABLE_BIS_D_COMPLIANT)
+						 EFUSE_ENABLE_BIS_D_COMPLIANT | \
+						 EFUSE_ENABLE_BIS_E_COMPLIANT)
 
 /*
  * EFUSE Secureboot configuration value in security word (0x3A)
@@ -157,11 +160,18 @@
 						EFUSE_ENABLE_BIS_COMPLIANT
 
 /*
- * EFUSE BIS rev D compliant enable configuaration for reg 0x3A:
+ * EFUSE BIS rev D compliant enable configuration for reg 0x3A:
  * BIT[26] - BIS rev D compliant enable = 1'b1
  */
 #define EFUSE_BIS_D_COMPLIANT_ENABLE		(EFUSE_BIS_COMPLIANT_ENABLE | \
 						 EFUSE_ENABLE_BIS_D_COMPLIANT)
+
+/*
+ * EFUSE BIS rev E compliant enable configuration for reg 0x3A:
+ * BIT[27] - BIS rev E compliant enable = 1'b1
+ */
+#define EFUSE_BIS_E_COMPLIANT_ENABLE		(EFUSE_BIS_COMPLIANT_ENABLE | \
+						 EFUSE_ENABLE_BIS_E_COMPLIANT)
 
 /*
  * EFUSE EROM TPM enable configuration for reg 0x3A:
@@ -213,7 +223,7 @@
 			uint32_t force_rec:1;
 			uint32_t bis_compliant:1;
 			uint32_t bis_d_compliant:1;
-			uint32_t spare:1;
+			uint32_t bis_e_compliant:1;
 			uint32_t sec_efuse_ver:4;
 			uint8_t sec_ver_num[SVN_EFUSE_SIZE];
 		} __packed;
@@ -401,6 +411,7 @@
 	GAUDI2_BIS_REV_B,
 	GAUDI2_BIS_REV_C,
 	GAUDI2_BIS_REV_D,
+	GAUDI2_BIS_REV_E,
 	GAUDI2_BIS_REV_MAX
 };
 
